tools/file_packager.py

```diff
@@ -649,7 +649,7 @@ def generate_js(data_target, data_files, metadata):
         var content = HEAPU32[start32++];
         var name = UTF8ToString(name_addr)
         // canOwn this data in the filesystem, it is a slice of wasm memory that will never change
-        Module['FS_createDataFile'](undefined, name, HEAP8.subarray(content, content + len), true, true, true);
+        Module['FS_createDataFile'](name, null, HEAP8.subarray(content, content + len), true, true, true);
```

Comment:
This seems odd to me too.  The way I handled this in #16050 I guess made it work both wasy:

```
-      var path = name ? PATH.join2(typeof parent === 'string' ? parent : FS.getPath(parent), name) : parent;
+      var path = name;
+      if (parent) {
+        parent = typeof parent === 'string' ? parent : FS.getPath(parent);
+        path = name ? PATH.join2(parent, name) : parent;
+      }
```